server/app/views/components/FieldWithLabel.java

```diff
@@ -583,8 +583,8 @@ private <T extends Tag<T> & IName<T> & IDisabled<T>> void generalApplyAttrsClass
         .withClasses(
             getFieldClasses(fieldTag, hasFieldErrors),
             // TODO(#5623): Use unified styles for disabled inputs
-            this.readOnly ? "text-gray-500" : "",
-            this.readOnly ? "bg-gray-100" : "")
+            this.readOnly ? "read-only:text-gray-500" : "",
+            this.readOnly ? "read-only:bg-gray-100" : "")
```

Comment:
Why: Conflict between `bg-white` and `bg-gray-100`. The `read-only` increases specificity when `input readonly="readonly"`. Including `text-gray-500` for to prevent the same issue down the road.